src/drivers/ins/eulernav_bahrs/eulernav_driver.cpp

```diff
@@ -0,0 +1,526 @@
+#include <matrix/Quaternion.hpp>
+#include <matrix/Euler.hpp>
+#include <atmosphere/atmosphere.h>
+
+EulerNavDriver::EulerNavDriver(const char *device_name)
+	: ModuleParams{nullptr}
+	, _serial_port{device_name, 115200, ByteSize::EightBits, Parity::None, StopBits::One, FlowControl::Disabled}
+	, _data_buffer{}
+	, _px4_accel{DRV_INS_DEVTYPE_BAHRS}
+	, _px4_gyro{DRV_INS_DEVTYPE_BAHRS}
+	, _attitude_pub{ORB_ID(vehicle_attitude)}
+	, _barometer_pub{ORB_ID(sensor_baro)}
+{
+	initialize();
+}
+
+EulerNavDriver::~EulerNavDriver()
+{
+	deinitialize();
+}
+
+int EulerNavDriver::task_spawn(int argc, char *argv[])
+{
+	int task_id = px4_task_spawn_cmd("bahrs", SCHED_DEFAULT, SCHED_PRIORITY_FAST_DRIVER,
+					 Config::TASK_STACK_SIZE, (px4_main_t)&run_trampoline, argv);
+
+	if (task_id < 0) {
+		_task_id = -1;
+		PX4_ERR("Failed to spawn task.");
+
+	} else {
+		_task_id = task_id;
+	}
+
+	return (_task_id < 0) ? 1 : 0;
+}
+
+EulerNavDriver *EulerNavDriver::instantiate(int argc, char *argv[])
+{
+	int option_index = 1;
+	const char *option_arg{nullptr};
+	const char *device_name{nullptr};
+
+	while (true) {
+		int option{px4_getopt(argc, argv, "d:", &option_index, &option_arg)};
+
+		if (EOF == option) {
+			break;
+		}
+
+		switch (option) {
+		case 'd':
+			device_name = option_arg;
+			break;
+
+		default:
+			break;
+		}
+	}
+
+	return new EulerNavDriver(device_name);
+}
+
+int EulerNavDriver::custom_command(int argc, char *argv[])
+{
+	return print_usage("unrecognized command");
+}
+
+int EulerNavDriver::print_usage(const char *reason)
+{
+	if (reason) {
+		PX4_WARN("%s\n", reason);
+	}
+
+	PRINT_MODULE_DESCRIPTION(
+		R"DESCR_STR(
+### Description
+
+Serial bus driver for the EULER-NAV Baro-Inertial AHRS.
+
+### Examples
+
+Attempt to start driver on a specified serial device.
+$ eulernav_bahrs start -d /dev/ttyS1
+Stop driver
+$ eulernav_bahrs stop
+)DESCR_STR");
+
+	PRINT_MODULE_USAGE_NAME("eulernav_bahrs", "driver");
+	PRINT_MODULE_USAGE_SUBCATEGORY("ins");
+	PRINT_MODULE_USAGE_COMMAND_DESCR("start", "Start driver");
+	PRINT_MODULE_USAGE_PARAM_STRING('d', nullptr, nullptr, "Serial device", false);
+	PRINT_MODULE_USAGE_COMMAND_DESCR("status", "Print driver status");
+	PRINT_MODULE_USAGE_COMMAND_DESCR("stop", "Stop driver");
+
+	return PX4_OK;
+}
+
+int EulerNavDriver::print_status()
+{
+	if (_is_initialized)
+	{
+		PX4_INFO("Elapsed time: %llu [us].\n", hrt_elapsed_time(&_statistics._start_time));
+		PX4_INFO("Total bytes received: %lu.\n", _statistics._total_bytes_received);
+		PX4_INFO("Inertial messages received: %lu. Navigation messages received: %lu.\n",
+			_statistics._inertial_message_counter, _statistics._navigation_message_counter);
+		PX4_INFO("Failed CRC count: %lu.\n", _statistics._crc_failures);
+
+	}
+	else
+	{
+		PX4_INFO("Initialization failed. The driver is not running.\n");
+	}
+
+	return PX4_OK;
+}
+
+void EulerNavDriver::run()
+{
+	_statistics._start_time = hrt_absolute_time();
+
+	while(false == should_exit())
```

Comment:
```suggestion
	while (!should_exit())
```
don't compare to true or false, fix throughout